common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RetryingBlockTransferor.java

```diff
@@ -149,6 +149,7 @@ public RetryingBlockTransferor(
    * in the event of transient IOExceptions.
    */
   public void start() {
+    currentListener = new RetryingBlockTransferListener();
```

Comment:
This is needed to spy `RetryingBlockTransferor` as, in the constructor, `this` still bounds to the unspied instance, and hence the inner class instance will also inherit it.

<img width="1120" alt="image" src="https://github.com/apache/spark/assets/17327104/232bc4f7-9608-43ae-acf6-5de658b5ed00">